jupyter_scheduler/extension.py

```diff
@@ -87,10 +95,12 @@ def initialize_settings(self):
             task_runner_run_interval=self.task_runner_run_interval,
         )
 
-        scheduler = self.scheduler_class(config=execution_config, task_runner_class=TaskRunner)
+        scheduler = self.scheduler_class(
+            config=execution_config, task_runner_class=self.task_runner_class
+        )
 
         self.settings.update(execution_config=execution_config, scheduler=scheduler)
 
-        if scheduler.task_runner:
+        if self.task_runner_run_interval > -1 and scheduler.task_runner:
```

Comment:
Well, this condition still allows for an invalid value like `-0.9`. Probably should be `<= 0`, since `0` also doesn't seem valid.